### PR TITLE
[diag] Implement diag stream command

### PR DESCRIPTION
**Pull request by parag-silabs**
- Add 'expect' simulation test.
- Update the diag/README.md

### CODE REVIEW

File: src/core/diags/factory_diags.cpp [hunk 1]
```diff
@@ -72,6 +72,8 @@ namespace FactoryDiags {
 const struct Diags::Command Diags::sCommands[] = {
     {"channel", &Diags::ProcessChannel},
     {"cw", &Diags::ProcessContinuousWave},
+    {"stream", &Diags::ProcessStream},
+    {"stream", &Diags::ProcessStream},
```
Comment (lmnotran) [inline src/core/diags/factory_diags.cpp:76]:
```suggestion
```
Remove this duplicate

**Review by lmnotran — CHANGES_REQUESTED** [review]
One minor change. Otherwise, it lGTM! 👍🏼

File: src/core/diags/factory_diags.cpp (hunk 1 above)
Comment (lmnotran) [reply src/core/diags/factory_diags.cpp:76]: Actually, can you shift this down a few lines so it's alphabetically sorted?

Comment (parag-silabs) [reply src/core/diags/factory_diags.cpp]: Make sense. Pushed the changes.

**Review by suveshpratapa — APPROVED** [review]
LGTM

File: src/core/diags/factory_diags.cpp [hunk 2]
```diff
@@ -573,6 +575,27 @@ Error Diags::ProcessContinuousWave(uint8_t aArgsLength, char *aArgs[], char *aOu
...
+Error Diags::ProcessStream(uint8_t aArgsLength, char *aArgs[], char *aOutput, size_t aOutputMaxLen)
+{
+    Error error = kErrorInvalidArgs;
+
+    VerifyOrExit(otPlatDiagModeGet(), error = kErrorInvalidState);
+    VerifyOrExit(aArgsLength > 0, error = kErrorInvalidArgs);
+
+    if (strcmp(aArgs[0], "start") == 0)
+    {
+        SuccessOrExit(error = otPlatDiagRadioTransmitStream(&GetInstance(), true));
```
Comment (hemanth-silabs) [inline src/core/diags/factory_diags.cpp:587]: Do you need SuccessOrExit here?

File: src/posix/platform/radio.cpp [hunk 3]
```diff
@@ -735,6 +735,20 @@ otError otPlatDiagRadioTransmitCarrier(otInstance *aInstance, bool aEnable)
...
 
+otError otPlatDiagRadioTransmitStream(otInstance *aInstance, bool aEnable)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError error;
+    char    cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+
+    snprintf(cmd, sizeof(cmd), "stream %s", aEnable ? "start" : "stop");
+    SuccessOrExit(error = sRadioSpinel.PlatDiagProcess(cmd, nullptr, 0));
```
Comment (hemanth-silabs) [inline src/posix/platform/radio.cpp:746]: Same here

**Review by hemanth-silabs — CHANGES_REQUESTED** [review]

File: src/core/diags/factory_diags.cpp (hunk 2 above)
Comment (parag-silabs) [reply src/core/diags/factory_diags.cpp:587]: It is not really needed, but also I see it's been defined like this in other diag CLI implementations and that could be for future proofing if we have added more code after #if..else block. Currently there is none so we can remove it. Let me know your thoughts.

**Review by hemanth-silabs — APPROVED** [review]
LGTM 👍

File: src/core/diags/factory_diags.cpp (hunk 2 above)
Comment (parag-silabs) [reply src/core/diags/factory_diags.cpp]: Remove SuccessOrExit check as it is not needed.

**Review by lmnotran — APPROVED** [review]

**Pull request merged**